qiskit_cold_atom/circuit_to_cold_atom.py

```diff
@@ -19,159 +19,179 @@
-                             - If the width of the circuit is too large
-                             - If the circuit has unbound parameters
+class CircuitTools:
+    """A class to provide tooling for cold-atomic circuits.
+
+    Since all methods are class methods this class does not need to be instantiated. This
+    class groups tools for cold atomic circuits. It also makes clear the ordering of the
+    fermionic wires that qiskit works with.
     """
 
-    if isinstance(circuits, QuantumCircuit):
-        circuits = [circuits]
-
-    # check for number of experiments allowed by the backend
-    if backend.configuration().max_experiments:
-        max_circuits = backend.configuration().max_experiments
-        if len(circuits) > max_circuits:
-            raise QiskitColdAtomError(
-                f"{backend.name()} allows for max. {max_circuits} different circuits; "
-                f"but {len(circuits)} circuits were given"
-            )
-    # check for number of individual shots allowed by the backend
-    if backend.configuration().max_shots and shots:
-        max_shots = backend.configuration().max_shots
-        if shots > max_shots:
-            raise QiskitColdAtomError(
-                f"{backend.name()} allows for max. {max_shots} shots per circuit; "
-                f"{shots} shots were requested"
-            )
-
-    for circuit in circuits:
-
-        try:
-            native_gates = {gate.name: gate.coupling_map for gate in backend.configuration().gates}
-            native_instructions = backend.configuration().supported_instructions
-        except NameError as name_error:
-            raise QiskitColdAtomError(
-                "backend needs to be initialized with config file first"
-            ) from name_error
-
-        if circuit.num_qubits > backend.configuration().num_qubits:
-            raise QiskitColdAtomError(
-                f"{backend.name()} supports circuits with up to "
-                f"{backend.configuration().num_qubits} wires, but"
-                f"{circuit.num_qubits} wires were given."
-            )
-
-        # If num_species is specified by the backend, the wires describe different atomic species
-        # and the circuit must exactly match the expected wire count of the backend.
-        if "num_species" in backend.configuration().to_dict().keys():
-            num_species = backend.configuration().num_species
-            if num_species > 1 and circuit.num_qubits < backend.configuration().num_qubits:
+    # Qiskit for fermions works with a sequential register definition. For fermionic
+    # modes with more than on species the circuits will have a corresponding number of
+    # sequential registers with the same length. For example, a three site system with two
+    # species will have two sequential registers with three wires each. Other packages may
+    # use an "interleaved" wire order.
+    __wire_order__ = "sequential"
```

Comment:
The only thing which bugs me a little bit is that this wire_order will now be included in the job payload of _any_ circuit built with qiskit_cold_atom, including the spin circuits. Maybe this is not too bad, since there we might also look at two different species at some point (e.g. Fred's simulator of gauge theories)

Reply:
I actually think that it is very good to have the wiring choice included in any circuit. Pure spin systems will fairly often fall back to bare `qiskit`and in almost any cold atom experiments with spinful bosons and fermions this question comes back.